### PR TITLE
Add docs on mTLS and Istio authorization policy

### LINKED ISSUE #3115: Document usage of mTLS in Knative
**Describe the change you'd like to see**
Add documentation for using Istio's mTLS policies with Knative, including the kinks of using it with activator.

/assign

### PR DESCRIPTION
**Pull request by arturenault**
Fixes #3115

## Proposed Changes 

- Add docs on mTLS and Istio authorization policy

/assign @nak3 @JRBANCEL

### CODE REVIEW

File: docs/serving/istio-authorization.md [hunk 1]
```diff
@@ -14,21 +14,83 @@ You must meet the following prerequisites to use Istio AuthorizationPolicy:
 - [Istio must be used for your Knative Ingress](https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-serving-component).
 - [Istio sidecar injection must be enabled](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/).
 
-## Enabling Istio AuthorizationPolicy
+## Mutual TLS in Knative
 
-For example, the following authorization policy denies all requests to workloads in namespace `serving-tests`.
+There are some important considerations when using mutual TLS in Knative; because Knative requests are frequently routed through activator, some considerations need to be made.
```
Comment (JRBANCEL) [inline docs/serving/istio-authorization.md:19]: Slightly redundant wording.

**Comment by arturenault** [comment]
Zhimin is oof and doesn't have doc approval 😅 

Trying @tcnghia

**Review by tcnghia — APPROVED** [review]

**Pull request merged**